app/scripts/services/quota.js

```diff
@@ -92,24 +92,66 @@ angular.module("openshiftConsole")
-            details = "You will not be able to create the " + humanizeKind(resource.kind) + " '" + resource.metadata.name + "'.";
+        if (!isNil(q.hard[type])) {
+          if (usageValue(q.hard[type]) <= usageValue(q.used[type])) {
+            var details;
+            if (resource.kind === 'Pod') {
+              details = "You will not be able to create the " + humanizeKind(resource.kind) + " '" + resource.metadata.name + "'.";
+            }
+            else {
+              details = "You can can still create " + humanizeKind(resource.kind) + " '" + resource.metadata.name + "' but no pods will be created until resources are freed.";
+            }
+            alerts.push({
+              type: resource.kind === 'Pod' ? 'error' : 'warning',
+              message: 'You are at your quota for ' + humanizeQuotaResource(type).toLowerCase() + ' on pods.',
+              details: details,
+              links: [{
+                href: "project/" + quota.metadata.namespace + "/quota",
+                label: "View Quota",
+                target: "_blank"
+              }]
+            });
           }
-          else {
-            details = "You can can still create " + humanizeKind(resource.kind) + " '" + resource.metadata.name + "' but no pods will be created until resources are freed.";
+          else if (type !== 'pods') {
+            var containerField;
+            switch (type) {
+              case 'cpu':
+              case 'requests.cpu':
+                containerField = "resources.requests.cpu";
+                break;
+              case 'limits.cpu':
+                containerField = "resources.limits.cpu";
+                break;
+              case 'memory':
+              case 'requests.memory':
+                containerField = "resources.requests.memory";
+                break;
+              case 'limits.memory':
+                containerField = "resources.limits.memory";
+                break;
+            }
+            var templateTotal = 0;
+            _.each(podTemplate.spec.containers, function(container) {
+              var templateVal = _.get(container, containerField);
+              if (templateVal) {
+                templateTotal += usageValue(templateVal);
+              }
+            });
+            if (usageValue(q.hard[type]) < usageValue(q.used[type]) + templateTotal) {
+              var detail;
+              if (resource.kind === 'Pod') {
+                detail = "You may not be able to create the " + humanizeKind(resource.kind) + " '" + resource.metadata.name + "'.";
+              }
+              else {
+                detail = "You can can still create " + humanizeKind(resource.kind) + " '" + resource.metadata.name + "' but you may not have pods created until resources are freed.";
+              }
+              alerts.push({
+                type: 'warning',
+                message: 'You are close to your quota for ' + humanizeQuotaResource(type).toLowerCase() + " on pods.",
```

Comment:
Will this lowercase `CPU`? might want a case option on `humanizeQuotaResource` like we have for `humanizeKind`.